repo/common/common.go

```diff
@@ -2,26 +2,62 @@ package common
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
+
+	"github.com/google/uuid"
 )
 
+// Find dynamic map key names passed with Parent["foo"] notation
+var bracketsRe = regexp.MustCompile(`\["([^\["\]]*)"\]`)
```

Comment:
This regular expression does not support escape sequences. I wouldn't advice writing your own quoted string parser. It's hard. I would advice using a JS parser library for parsing the ENTIRE glob, e.g. https://pkg.go.dev/github.com/robertkrimen/otto/parser, and then work with the AST.

Reply:
Alternatively, you can use the Go language parser for the glob. It has a different set of escape sequences than JS, but is still better than rolling your own.